pkg/cvo/cvo.go

```diff
@@ -37,6 +37,9 @@ import (
 	"github.com/openshift/cluster-version-operator/pkg/cvo/internal/dynamicclient"
 	"github.com/openshift/cluster-version-operator/pkg/internal"
 	"github.com/openshift/cluster-version-operator/pkg/payload"
+	"github.com/openshift/cluster-version-operator/pkg/payload/preflight"
```

Comment:
> I expected this PR to summarize an Upgradeable condition - ideally before a user ever upgrades.

That would be a long running syncer, and I don't think it's great to add that at this time.

> The current structure can't easily do that (and there are advantages to having preflight check coming on the new code),

The current structures goal is to prevent upgrades, the long running syncer doesn't do that.. even if we add the long running syncer, we still needs this code to check the precondition before cvo accepts the upgrade. So I think the syncer to provide the Upgradeable condition is _orthogonal_ to this, and perhaps when we add that later on, the precondition could use that as another check??

Reply:
^^ ping @smarterclayton

Reply:
> That would be a long running syncer, and I don't think it's great to add that at this time.

I don't understand why.  You just have to summarize the state.

Blocking without reporting seems unacceptable to me.  I should know up top whether upgrades should even work.

I think the miscommunication / gap here is that we already have an implemented mechanism for reporting "Don't upgrade me" as Upgradable false since 4.1 GA, but this is targeting other checks which are much more specific / tied to implementation.  We may need to have a higher bandwidth conversation on that, but the generic condition is better at isolating CVO from details, and behaves exactly like degraded in that sense.